### PR TITLE
add query for clinical type & createdAt

### PR DESCRIPTION
**Pull request by jaserud**
**Description of changes**

- added a query to ask clinical for clinical types list
- also take actual createdAt from clinical entity
- also also allow null values in clinical records
- changed shortName to programShortName for clinical submissions

Note: merge after https://github.com/icgc-argo/argo-clinical/pull/173

### CODE REVIEW

File: schemas/Clinical/index.js [hunk 1]
```diff
@@ -293,7 +301,7 @@ const convertClinicalSubmissionEntityToGql = (type, entity) => {
       get(entity, 'dataErrors', []).map(error => convertClinicalSubmissionErrorToGql(error)),
     dataUpdates: () =>
       get(entity, 'dataUpdates', []).map(update => convertClinicalSubmissionUpdateToGql(update)),
-    createdAt: new Date(), // this is a place holder for now
+    createdAt: entity.createdAt || null,
```
Comment (hlminh2000) [inline schemas/Clinical/index.js]: is `entity.createdAt` a `Date` instance?

Comment (jaserud) [reply schemas/Clinical/index.js]: yes, it is a Date instance in [clinical](https://github.com/icgc-argo/argo-clinical/blob/86c8626ed4e95804eba485ed74db33b2238db13f/src/submission/submission-entities.ts#L156)

Comment (hlminh2000) [reply schemas/Clinical/index.js]: I think we'd be losing that context through the JSON serialization though :/

Comment (hlminh2000) [reply schemas/Clinical/index.js]: maybe `entity.createdAt ? new Date(entity.createdAt) : null` to be safe? :P

**Pull request merged**